Add docs to mention changes regarding modular upgrades (#5932)
aws/eks-anywhere: docs/content/en/docs/tasks/cluster/cluster-upgrades/baremetal-upgrades.md

>**_NOTE:_** Currently only Minor Version Upgrades are support for Bare Metal clusters. No other aspects of the cluster upgrades are currently supported.
>

> ### Modular Cluster Upgrades

Reviewer: I don't think we need a separate section for this. Instead I would like to call out in a "Note" that mentions previous versions of EKS-A (v0.13 and below?) had a bug that could cause workload clusters to upgrade when a management cluster EKS-A version upgrade was triggered. Now that no longer happens.

To allow users to be explicit when deciding which clusters to upgrade, workload clusters are not affected by management cluster upgrades.
Therefore, each cluster must be individually upgraded to enjoy the latest features on all clusters.

> When triggering a workload cluster upgrade after upgrading the management cluster, it will not only reflect the changes in the cluster spec, but also any changes derived from the difference in business logic between the old and new EKS-A cluster controllers.

Reviewer 2: Trying to make this info a bit more accessible to readers with different backgrounds:
```suggestion
When triggering a workload cluster upgrade after upgrading the management cluster, it will not only reflect your changes in the cluster spec, but also any new improvements included in the new EKS-A controller.
```

> When triggering a workload cluster upgrade after upgrading the management cluster, it will not only reflect the changes in the cluster spec, but also any changes derived from the difference in business logic between the old and new EKS-A cluster controllers.

Reviewer: This is very vague ...

> This difference in logic can trigger a machine rollout, even if the updates to the cluster spec didn't require one (for example, scaling down a worker node group).

Reviewer: "can trigger a machine rollout" ... we need to be more specific if possible

Reviewer: What's our position here?

### EKS Anywhere Version Upgrades

We encourage that you stay up to date with the latest EKS Anywhere version, as new features, bug fixes, or security patches might be added in each release. You can find the content, such as supported OS versions and changelog, of each EKS Anywhere version on the [What’s New]({{< relref "../../../reference/changelog/" >}}) page.

Download the [latest or target EKS Anywhere release](https://github.com/aws/eks-anywhere/releases/) and run `eksctl anywhere upgrade cluster` command to upgrade a cluster to a specific EKS Anywhere version.

> We provide a maximum skew version support of one EKS Anywhere minor version for management and workload clusters.

Reviewer: We need to add a comment on what happens if a workload cluster creation starts on v0.15.x, then the customer upgrades management cluster to v0.16. Does it work? Does it break?

> This means the management cluster can only be one EKS Anywhere minor version after the workload clusters (such as v0.16 and v0.17).

Reviewer 2: ```suggestion
This means the management cluster can only be one EKS Anywhere minor version newer the workload clusters (such as v0.16 for workload clusters if the management cluster is at v0.17).
```

In the event that you want to upgrade your management cluster to a version that does not satisfy this condition, we suggest upgrading the workload cluster's EKS-A version first, followed by upgrading to your desired EKS-A version for the management cluster.

> **Skipping Amazon EKS Anywhere minor versions during cluster upgrade (such as going from v0.14 directly to v0.16) is NOT allowed.** EKS Anywhere team performs regular upgrade reliability testing for sequential version upgrade (e.g. going from version 0.14 to 0.15, then from version 0.15 to 0.16), but we do not perform testing on non-sequential upgrade path (e.g. going from version 0.14 directly to 0.16). You should not skip minor versions during cluster upgrade. However, you can choose to skip patch versions.

Reviewer: This is not accurate. I want to soften this language. We aren't doing anything in the product to prevent a customer from skipping EKS-A minor versions when they upgrade. We should position it as "do it at your own risk"

>**_NOTE:_** Currently only Minor Version Upgrades are support for Bare Metal clusters. No other aspects of the cluster upgrades are currently supported.
>

{{% alert title="Important" color="warning" %}}

> In `eksctl anywhere` version `v0.15.0`, we introduced the full lifecycle controller to fully manage new workload clusters.

Reviewer: We have v0.15.0 here but v0.13.0 in the vsphere/cloudstack upgrade page. Which one is it?

> The controller behavior would roll out new nodes in the workload cluster whenever the user upgraded the management cluster and its management components.

Reviewer 3: ```suggestion
In this version, the controller rolls out new nodes in the workload cluster whenever the user upgrades the management cluster and its management components.
```

In `eksctl anywhere` version `v0.16.0`, we changed this behavior to allow users to be explicit when deciding which clusters to upgrade.
Therefore, workload clusters are no longer affected by management cluster upgrades.
Due to this change, each cluster must be individually upgraded to enjoy the latest features on all clusters.
If you have a management cluster running EKS Anywhere version v0.15, you can successfully upgrade to EKS Anywhere version v0.16 and observe no changes to any of its workload clusters.

> When triggering a workload cluster upgrade after upgrading the management cluster, please keep in mind that it will not only reflect your changes in the cluster spec, but also any new improvements included in the new EKS-A controller.

Reviewer 3: ```suggestion
When triggering a workload cluster upgrade after upgrading the management cluster, please keep in mind that it will not only reflect your changes in the cluster spec, but also any new improvements included in the new EKS Anywhere controller.
```
